src/layer-model.js

```diff
@@ -32,6 +50,42 @@ class LayerModel {
       }
     });
   }
+
+  static parse(layerSpec = {}) {
+    const {
+      params_config: paramsConfig,
+      sqlParams_config: sqlParamsConfig,
+      decode_config: decodeConfig,
+      body
+    } = layerSpec.layerConfig;
+    const { url } = body;
+    let params = {};
+    const sqlParams = {};
+    const decodeParams = {};
+
+    (paramsConfig || []).forEach(_param => {
```

Comment:
why not use `.reduce` in these cases to make it blazin 🔥

Reply:
please